src/tasks/escalate.rs

```diff
@@ -16,90 +20,136 @@ pub async fn escalate_txs(app: Arc<App>) -> eyre::Result<()> {
-                .send_transaction(TypedTransaction::Eip1559(eip1559_tx), None)
-                .await;
-
-            let pending_tx = match pending_tx {
-                Ok(pending_tx) => {
-                    tracing::info!(?pending_tx, "Tx sent successfully");
-                    pending_tx
-                }
-                Err(err) => {
-                    tracing::error!(error = ?err, "Failed to send tx");
-                    continue;
-                }
-            };
-
-            let tx_hash = pending_tx.tx_hash();
-
-            app.db
-                .escalate_tx(
-                    &tx.id,
-                    tx_hash,
-                    max_fee_per_gas,
-                    max_priority_fee_per_gas,
-                )
-                .await?;
-
-            tracing::info!(id = ?tx.id, hash = ?tx_hash, "Tx escalated");
+        for (relayer_id, txs) in txs_for_escalation {
+            futures.push(escalate_relayer_txs(&app, relayer_id, txs));
+        }
+
+        while let Some(result) = futures.next().await {
+            if let Err(err) = result {
+                tracing::error!(error = ?err, "Failed escalating txs");
+            }
         }
 
         tokio::time::sleep(app.config.service.escalation_interval).await;
     }
 }
+
+async fn escalate_relayer_txs(
+    app: &App,
+    relayer_id: String,
+    txs: Vec<TxForEscalation>,
+) -> eyre::Result<()> {
+    let relayer = app.db.get_relayer(&relayer_id).await?;
+
+    for tx in txs {
+        tracing::info!(id = tx.id, tx.escalation_count, "Escalating tx");
+
+        if !should_send_transaction(&app, &relayer).await? {
+            tracing::warn!(id = tx.id, "Skipping transaction broadcast");
+
+            return Ok(());
+        }
+
+        let escalation = tx.escalation_count + 1;
+
+        let middleware = app
+            .signer_middleware(tx.chain_id, tx.key_id.clone())
+            .await?;
+
+        let fees = app
+            .db
+            .get_latest_block_fees_by_chain_id(tx.chain_id)
+            .await?
+            .context("Missing block")?;
+
+        // Min increase of 20% on the priority fee required for a replacement tx
+        let factor = U256::from(100);
+        let increased_gas_price_percentage =
+            factor + U256::from(20 * (1 + escalation));
+
+        let max_fee_per_gas_increase = tx.initial_max_fee_per_gas.0
+            * increased_gas_price_percentage
+            / factor;
+
+        let max_fee_per_gas =
+            tx.initial_max_fee_per_gas.0 + max_fee_per_gas_increase;
+
+        let max_priority_fee_per_gas =
+            max_fee_per_gas - fees.fee_estimates.base_fee_per_gas;
+
+        tracing::warn!(
+            "Initial tx fees are max = {}, priority = {}",
+            tx.initial_max_fee_per_gas.0,
+            tx.initial_max_priority_fee_per_gas.0
+        );
+        tracing::warn!("Escalating with max fee = {max_fee_per_gas} and max priority = {max_priority_fee_per_gas}");
```

Comment:
Since relayers can escalate transactions concurrently, it would be helpful to know which transaction id these fees are for.

We should consider using `info!` here instead of `warn!` since escalation is not necessarily unexpected.

Additionally, we can condense this into one log.
```rust
tracing::info!(tx_id = tx.id, previous_mfpg = tx.initial_max_fee_per_gas.0, updated_mfpg = max_fee_per_gas, previous_mpfpg = tx.initial_max_priority_fee_per_gas.0, updated_mpfpg = max_priority_fee_per_gas);
```

Reply:
Yeah, I want to improve on logging in the next PR as it'll also affect code not touched by this PR